iceaxe/__tests__/test_session.py

```diff
@@ -907,3 +913,49 @@ async def test_for_update_of_with_join(db_connection: DBConnection):
                 pytest.fail("Should have raised an error")
         finally:
             await other_conn.conn.close()
+
+
+@pytest.mark.asyncio
+async def test_select_same_column_name_from_different_tables(
+    db_connection: DBConnection,
+):
+    """
+    Test that we can correctly select and distinguish between columns with the same name
```

Comment:
Switching to our new `sql` selection internally also fixed an issue where individual field selections with the name column would conflict with one another during sql->TableBase conversion.